[Evan Loo] iP

Evande1 opened the pull request:
## MILK 
> "Wow, that was really cool." - Dani, on Karlson, probably. ([source](https://www.youtube.com/c/DaniDev))

Bad memory? Lactose intolerant? Milk prevents constipation intellectually and physically! It's, 
- Lactose-free
- Text-based
- Cross platform

All you need to do is,
1. Buy some Sparkle Slime Balloons 💦 
2. Inhale copium 
3. Wishful thinking

And it is ~~**FREE**~~ your net worth and more.

Features:
- [x] Works
- [ ] Breaks (Soon)
- [ ] Checkstyle (In the unforeseeable future)
 
If you get excited by this code, you can practice codeine too! Here's the ```main``` method:
```java
public class Main {
    public static void main(String[] args) {
        System.out.println("Hello World");
    }
}
```

Commit:
Add Gradle support

Evande1 committed:
Level-1

Evande1 committed:
Level-2

Evande1 committed:
Level-3

Evande1 committed:
Level-4

Evande1 committed:
A-TextUiTesting

Evande1 committed:
Level-5

Evande1 committed:
Level-6

Evande1 committed:
Level-7

Evande1 committed:
Merge branch 'branch-Level-7'

Evande1 committed:
Level-7

Evande1 committed:
Merge branch 'branch-Level-7'

Evande1 committed:
Merge pull request #1 from Evande1/branch-Level-7

Branch level 7

Evande1 committed:
Merge branch 'master' of https://github.com/Evande1/CS2103T-ip

Evande1 committed:
Level-8

Evande1 committed:
Merge branch 'branch-Level-8'

Evande1 committed:
A-MoreOOP

Evande1 committed:
A-Packages

Evande1 committed:
Merge branch 'add-gradle-support'

Evande1 committed:
A-Gradle

Evande1 committed:
A-JUnit

Evande1 committed:
A-Jar

Evande1 committed:
A-JavaDoc

Evande1 committed:
Merge branch 'branch-A-JavaDoc'

Evande1 committed:
A-CodingStandard

Evande1 committed:
merge conflict

Evande1 committed:
DukeJavaDoc

Evande1 committed:
Duke

Evande1 committed:
Merge branch 'branch-A-JavaDoc'

Evande1 committed:
minor tweaks

Evande1 committed:
merge conflict

Evande1 committed:
Level-9

Evande1 committed:
merge conflict

bryanhce commented on src/main/java/duke/command/AddDeadlineCommand.java:
```diff
@@ -0,0 +1,43 @@
+    private LocalDateTime by;
+
+    /**
+     * Constructor for AddDeadlineCommand
```
Should there be an empty line between the first line description and the parameter section?

bryanhce commented on src/main/java/duke/command/AddDeadlineCommand.java:
```diff
@@ -0,0 +1,43 @@
+
+    /**
+     * Executes the command
+     * @param tasks TaskList object that stores tasks
```
I noticed the same format of the missing line in other several other places too.

bryanhce commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,29 @@
+    /**
+     * Executes the command
+     * @param tasks TaskList object that stores tasks
+     * @param ui Ui object deals with user interaction
```
Should there be a punctuation behind each parameter description?

bryanhce commented on src/main/java/duke/command/ExitCommand.java:
```diff
@@ -0,0 +1,33 @@
+     *
+     * @return true for program to exit
+     */
+    @Override
```
I am not 100% sure but this isExit() method is very similar to that of the isExit() method in Command class, maybe instead of writing JavaDocs, you can use the @inheritDoc tag?

bryanhce commented on src/main/java/duke/main/Duke.java:
```diff
@@ -0,0 +1,57 @@
+ * Represents the Duke programme. Carries out the commands and handles exceptions
+ */
+public class Duke {
+    private static final String FILEPATH = "data/duke.txt";
```
I like how you made this a constant so that it will be easier to change in the future.

bryanhce commented on src/main/java/duke/main/Parser.java:
```diff
@@ -0,0 +1,172 @@
+        return splitReply[0].equals("find");
+    }
+
+    private static int validateIndex(String userReply) throws DukeException {
```
I like how comprehensive the input validation is!

bryanhce commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,66 @@
+     */
+    @Override
+    public String toString() {
+        StringBuilder str = new StringBuilder();
```
I like how you used StringBuilder class instead of String for you string manipulation.

bryanhce commented on src/main/java/duke/task/Events.java:
```diff
@@ -0,0 +1,37 @@
+/**
+ * Represents an Events task. A Event object contains the description and when event is happening
+ */
+public class Events extends Task {
```
Should the name of the class be singular instead of plural?

bryanhce commented on src/main/java/duke/task/ToDos.java:
```diff
@@ -0,0 +1,33 @@
+/**
+ * Represents a ToDo task. A ToDo object contains the description of task.
+ */
+public class ToDos extends Task {
```
Should the name of the class be singular instead of plural? Conversely, if you would like to standardise the classes, you might want to change the Deadline class to its plural form as well.

bryanhce reviewed (commented):
LGTM. The code was written very well and comprehensively. Some nits in the JavaDocs and naming conventions to fix. Nice job!

Evande1 committed:
Level-10

Evande1 committed:
Merge branch 'branch-Level-10'

Evande1 committed:
A-Assertions

Added assertion in Duke initialisation as there should be no file
parsing error messages.

Conditions are already handled within UI.

Removed printing of statements in terminal by UI as GUI should be the
one printing the duke replies. Troubleshooting is also done so there is
no need for printing of data.

Evande1 committed:
Merge pull request #2 from Evande1/branch-A-Assertions

A-Assertions

Evande1 committed:
A-CodeQuality

Add a function in Storage to make the code more readable.

Evande1 committed:
Merge branch 'master' into branch-A-CodeQuality

Evande1 committed:
Merge pull request #3 from Evande1/branch-A-CodeQuality

A-CodeQuality

Evande1 committed:
refactoring and additional function

Evande1 committed:
Merge branch 'branch-A-CodeQuality' of https://github.com/Evande1/CS2103T-ip into branch-A-CodeQuality

Evande1 committed:
Merge branch 'branch-A-CodeQuality'

Evande1 committed:
Merge branch 'master' of https://github.com/Evande1/CS2103T-ip

Evande1 committed:
A-Extension

Find next available timing, but might need improvements

Evande1 committed:
Merge pull request #4 from Evande1/testing

Testing

Evande1 committed:
changed runtest

Evande1 committed:
Merge branch 'testing'

Evande1 committed:
test

Evande1 committed:
Merge branch 'master' of https://github.com/Evande1/CS2103T-ip

Evande1 committed:
Create gradle.yml

Evande1 committed:
Update gradle.yml

Evande1 committed:
Merge pull request #5 from Evande1/Evande1-patch-1

Github Actions CI

Evande1 committed:
minor tweaks

Evande1 committed:
Merge branch 'branch-Addon'

Evande1 committed:
Merge pull request #7 from Evande1/branch-Addon

minor tweaks

Evande1 committed:
upload pictures

Evande1 committed:
Merge branch 'master' of https://github.com/Evande1/CS2103T-ip

Evande1 committed:
UserGuide

Evande1 committed:
Update README.md

Evande1 committed:
Update README.md

Evande1 committed:
release

Evande1 committed:
Merge branch 'master' of https://github.com/Evande1/CS2103T-ip

Evande1 committed:
Update README.md
